docs/visual_studio_code.md

```diff
@@ -0,0 +1,269 @@
+    title: str
+    knight: Knight
+
+
+quest = Quest(
+    title='To seek the Holy Grail',
+    knight={'title': 'Sir Lancelot', 'age': 23}
+)
+```
+
+The type of the field `knight` is declared with the class `Knight` (a *pydantic* model) and the code is passing a literal `dict` instead. This is still valid for *pydantic*, and the `dict` would be automatically converted to a `Knight` instance.
+
+Nevertheless, it would be detected as a type error:
+
+![VS Code strict type errors with model](./img/vs_code_07.png)
+
+In those cases, there are several ways to disable or ignore strict errors in very specific places, while still preserving them in the rest of the code.
+
+Below are several techniques to achieve it.
+
+#### Disable type checks in a line
+
+You can disable the errors for a specific line using a comment of:
+
+```
+# type: ignore
+```
+
+coming back to the example with `age='23'`, it would be:
+
+```Python hl_lines="10"
+from pydantic import BaseModel
+
+
+class Knight(BaseModel):
+    title: str
+    age: int
+    color: str = 'blue'
+
+
+lancelot = Knight(title='Sir Lancelot', age='23')  # type: ignore
+```
+
+that way Pylance and mypy will ignore errors in that line.
+
+**Pros**: it's a simple change in that line to remove errors there.
+
+**Cons**: any other error in that line will also be omitted, including type checks, misspelled arguments, required arguments not provided, etc.
+
+#### Override the type of a variable
+
+You can also create a variable with the value you want to use and declare it's type explicitly with `Any`.
+
+```Python hl_lines="1 11-12"
+from typing import Any
+from pydantic import BaseModel
+
+
+class Knight(BaseModel):
+    title: str
+    age: int
+    color: str = 'blue'
+
+
+age_str: Any = '23'
+lancelot = Knight(title='Sir Lancelot', age=age_str)
+```
+
+that way Pylance and mypy will interpret the variable `age_str` as if they didn't know its type, instead of knowing it has a type of `str` when an `int` was expected (and then showing the corresponding error).
+
+**Pros**: errors will be ignored only for a specific value, and you will still see any additional errors for the other arguments.
+
+**Cons**: it requires importing `Any` and a new variable in a new line for each argument that needs ignoring errors.
+
+#### Override the type of a value with `cast`
+
+The same idea from the previous example can be put on the same line with the help of `cast()`.
+
+This way, the type declaration of the value is overriden inline, without requiring another variable.
+
+```Python hl_lines="1 11"
+from typing import Any, cast
+from pydantic import BaseModel
+
+
+class Knight(BaseModel):
+    title: str
+    age: int
+    color: str = 'blue'
+
+
+lancelot = Knight(title='Sir Lancelot', age=cast(Any, '23'))
+```
+
+`cast(Any, '23')` doesn't affect the value, it's still just `'23'`, but now Pylance and mypy will assume it is of type `Any`, which means, they will act as if they didn't know the type of the value.
```

Comment:
Just a small remark @tiangolo
Why not say `cast(int, '23')` ?

Reply:
Thanks for the review, update to the PR, and the note @PrettyWood!

---

My intention with using `cast(Any, '23')` was to make it something simple to copy-paste for other use cases and types.

For example, if it was something like a `Dict[str, List[SomeModel]]` and the value was a dictionary with lists of dictionaries (instead of Pydantic models), e.g. parsed from some JSON, duplicating the annotation from the original field would be a bit cumbersome, would require duplication (which is always problematic and I think avoiding duplication is one of the greatest advantages of Pydantic), and adding the full type annotation/cast wouldn't provide much more help as it's only for mypy/Pylance.

Also, the idea was to make it explicit that this is just to "trick" mypy and Pylance to not pay attention to the actual types. And to avoid any possible confusion from users seeing a `cast(int, '23')`, which is technically wrong (in terms of types).

And also to prevent them from thinking that they can just copy that anywhere else and assume it would work (e.g. thinking it would convert the types automatically or something). At least if they use `cast(Any, '23')` in any other place, they lose autocompletion and can (hopefully) realize that the value is not an actual integer.

Reply:
Makes sense! Thanks 👍